### PR TITLE
all: remove ethash pow, only retain shims needed for consensus and tests

### PR DESCRIPTION
**Pull request by karalabe**
Remove PoW 😅 

The PR also removes the ethash related flags, configs and API methods in a non-backwards compatible way. There is quite a bit of stuff that we'd need to keep dangling in the code to avoid "that one user" not having upgrade issues. As we're right after a fork, there's no immediate urgency to update, might as well bork things now.

~~Probably the PR should add some vocal failure modes for running Geth in non-beacon-ethash mode.~~ Done!

We also need a followup PR to move the consensus things into a `legacy` or `historical` package (or directly into the `beacon` package) and to rewrite the tests to use some beacon shims instead of the current skeletons in `ethash`. Then we can be free. That needs a bit of refactor so I wanted to start by mass removes and then review the refactors separately more easily.

### CODE REVIEW

**Comment by sandakersmann** [comment]
You also have some residue in the lower parts of README.md about running a private miner.

File: cmd/evm/internal/t8ntool/flags.go [hunk 1]
```diff
@@ -125,14 +125,6 @@ var (
...
 	}
-	SealEthashFlag = &cli.BoolFlag{
-		Name:  "seal.ethash",
-		Usage: "Seal block with ethash.",
-	}
-	SealEthashDirFlag = &cli.StringFlag{
-		Name:  "seal.ethash.dir",
-		Usage: "Path to ethash DAG. If none exists, a new DAG will be generated.",
-	}
 	SealEthashModeFlag = &cli.StringFlag{
```
Comment (rjl493456442) [inline cmd/evm/internal/t8ntool/flags.go:128]: This flag is also not used, please remove it.

Comment (karalabe) [reply cmd/evm/internal/t8ntool/flags.go]: Fixed.

File: miner/worker_test.go [hunk 2]
```diff
@@ -208,10 +208,6 @@ func newTestWorker(t *testing.T, chainConfig *params.ChainConfig, engine consens
 	return w, backend
 }
 
-func TestGenerateBlockAndImportEthash(t *testing.T) {
-	testGenerateBlockAndImport(t, false)
```
Comment (MariusVanDerWijden) [inline miner/worker_test.go:212]: Shouldn't we also remove the functionality in the test? L 229ff

**Comment by zhiqiangxu** [comment]
Does this mean `go-ethereum` doesn't support sync from genesis any more?

**Comment by karalabe** [comment]
It can still sync from genesis, but since the PoW part is frozen and can
never change any more, verifying the PoW fields are skipped.

**Review by tuanggo — COMMENTED** [review]
opem

**Review by tuanggo — COMMENTED** [review]
ok

**Pull request merged**